src/test/java/com/thebois/utils/MatrixUtilsTests.java

```diff
@@ -64,22 +61,18 @@ public void forEachElementGetsAllElementsInOrder(final Integer[][] matrix,
 
     @ParameterizedTest
     @MethodSource("getMatricesToTest")
-    public void matrixToCollectionTest(final Optional<String>[][] matrix,
+    public void matrixToCollection(final String[][] matrix,
```

Comment:
style

You're just stating the name of the method you're testing. Describe what is being tested, e.g. `toCollectionTurnsMatrixIntoExpectedCollectionOfItems` or something similar.